bootloader/src/pyi_launch.c

```diff
@@ -204,16 +206,17 @@ _extract_dependency(ARCHIVE_STATUS *archive_pool[], const char *item)
     }
 
     pyi_path_dirname(dirname, path);
+    pyi_path_dirname(homepath_parent, archive_status->homepath);
+    char * contents_directory = pyi_arch_get_option(archive_pool[0], "pyi-meipass-prefix");
```

Comment:
For the sake of consistency, we should probably catch the case when the option is not set.

Reply:
Consistency? Would this not result in a null pointer deref?

Reply:
> Consistency? Would this not result in a null pointer deref?

It would. But I imagine that if the option is not set on the referenced binary, it likely was not set on the original binary, either. Although either situation is likely a corner case, as the python-side code currently always sets the option.

Reply:
Or rather, `archive_pool[0]` where we are reading the option from is the original binary (= "self"), not the first referenced binary. So the check here should actually not be necessary (as the check in the other codepath should guarantee that the option is set).

Reply:
By catch, you just mean throw an error, right? Not invent a default?

Reply:
Yes, raise an error.